components/script/dom/htmlinputelement.rs

```diff
@@ -1704,6 +1703,38 @@ fn broadcast_radio_checked(broadcaster: &HTMLInputElement, group: Option<&Atom>)
     }
 }
 
+fn perform_radio_group_validation(elem: &HTMLInputElement, group: Option<&Atom>, can_gc: CanGc) {
+    for r in radio_group_iter(elem, group) {
+        r.validity_state()
+            .perform_validation_and_update(ValidationFlags::all(), can_gc);
+    }
+}
+
+fn update_related_validity_states(elem: &HTMLInputElement, can_gc: CanGc) {
+    match elem.input_type() {
+        InputType::Radio => {
+            perform_radio_group_validation(elem, elem.radio_group_name().as_ref(), can_gc)
+        },
+        _ => {
+            elem.validity_state()
+                .perform_validation_and_update(ValidationFlags::all(), can_gc);
+        },
+    }
+}
+
+fn find_related_radios<'a>(
+    context: &'a UnbindContext,
+    elem: &'a HTMLInputElement,
+    form: Option<&'a HTMLFormElement>,
+    group: Option<&'a Atom>,
+) -> impl Iterator<Item = DomRoot<HTMLInputElement>> + 'a {
+    let root = context.parent.GetRootNode(&GetRootNodeOptions::empty());
```

Comment:
Instead of duplicate radio_group_iter, can we make `radio_group_iter` call `find_related_radios`? This requires passing a `&'a Node` argument instead of `context`.